### PR TITLE
Apply clang-tidy to more code and do some related cleanups

### PR DESCRIPTION
**Pull request by AJenbo**

### CODE REVIEW

File: Source/inv.cpp [hunk 1]
```diff
@@ -625,52 +619,58 @@ bool AutoEquipEnabled(const PlayerStruct &player, const ItemStruct &item)
...
 
-		for (int i = 10; i < 19 && !done; i++) {
-			done = AutoPlaceItemInInventorySlot(player, i, item, persistItem);
+		for (int i = 10; i < 19; i++) {
+			if (AutoPlaceItemInInventorySlot(player, i, item, persistItem))
+				return true;
 		}
-		return done;
+		return false;
 	}
```
Comment (thebigMuh) [inline Source/inv.cpp:675]: Might want to add an assert in here that this is an unhandled item size.

File: Source/inv.cpp [hunk 2]
```diff
@@ -686,148 +686,106 @@ bool AutoPlaceItemInInventory(PlayerStruct &player, const ItemStruct &item, bool
...
 }
 
 int SwapItem(ItemStruct *a, ItemStruct *b)
 {
-	ItemStruct h;
-
-	h = *a;
+	ItemStruct h = *a;
 	*a = *b;
 	*b = h;
```
Comment (thebigMuh) [inline Source/inv.cpp:790]: std::swap(*a, *b);

File: Source/inv.cpp [hunk 3]
```diff
@@ -836,19 +794,14 @@ int SwapItem(ItemStruct *a, ItemStruct *b)
...
-	sy = icursH28;
-	done = false;
-	for (r = 0; (DWORD)r < NUM_XY_SLOTS && !done; r++) {
+	int i = mx + (icursW / 2);
+	int j = my + (icursH / 2);
+	int sx = icursW28;
+	int sy = icursH28;
+	bool done = false;
+	int r = 0;
+	for (; (DWORD)r < NUM_XY_SLOTS && !done; r++) {
```
Comment (thebigMuh) [inline Source/inv.cpp:804]: The (DWORD) is bonkers. None of the values involved is a DWORD. And it's a C style cast.

File: Source/inv.cpp [hunk 4]
```diff
@@ -1219,12 +1172,10 @@ void CheckInvPaste(int pnum, int mx, int my)
 
 void CheckInvSwap(int pnum, BYTE bLoc, int idx, uint16_t wCI, int seed, bool bId, uint32_t dwBuff)
 {
-	PlayerStruct *p;
-
 	memset(&items[MAXITEMS], 0, sizeof(*items));
 	RecreateItem(MAXITEMS, idx, wCI, seed, 0, (dwBuff & CF_HELLFIRE) != 0);
 
-	p = &plr[pnum];
+	PlayerStruct *p = &plr[pnum];
```
Comment (thebigMuh) [inline Source/inv.cpp:1178]: PlayerStruct &p = plr[pnum];

File: Source/inv.cpp [hunk 5]
```diff
@@ -1999,8 +1941,8 @@ char CheckInvHLight()
 
 	int8_t rv = -1;
 	infoclr = UIS_SILVER;
-	pi = nullptr;
-	p = &plr[myplr];
+	ItemStruct *pi = nullptr;
+	PlayerStruct *p = &plr[myplr];
```
Comment (thebigMuh) [inline Source/inv.cpp:1945]: PlayerStruct &p = plr[myplr];

File: Source/stores.cpp [hunk 6]
```diff
@@ -183,26 +180,26 @@ void PrintStoreItem(ItemStruct *x, int l, uint16_t flags)
...
 		strcat(sstr, _("Indestructible,  "));
 	}
 	if (x->_itype == ITYPE_MISC)
 		sstr[0] = '\0';
-	str = x->_iMinStr;
-	mag = x->_iMinMag;
-	dex = x->_iMinDex;
+	int8_t str = x->_iMinStr;
+	uint8_t mag = x->_iMinMag;
+	int8_t dex = x->_iMinDex;
```
Comment (thebigMuh) [inline Source/stores.cpp:193]: Is there an actual reason for the mixed types?

File: Source/stores.cpp [hunk 7]
```diff
@@ -457,7 +449,7 @@ void S_StartSSell()
 			if (storehold[storenumh]._iMagical != ITEM_QUALITY_NORMAL && storehold[storenumh]._iIdentified)
 				storehold[storenumh]._ivalue = storehold[storenumh]._iIvalue;
 
-			if (!(storehold[storenumh]._ivalue >>= 2))
+			if ((storehold[storenumh]._ivalue >>= 2) == 0)
```
Comment (thebigMuh) [inline Source/stores.cpp:452]: / 4

**Review by thebigMuh — APPROVED** [review]
Didn't nitpick too much, seems fine.

File: Source/gmenu.cpp [hunk 8]
```diff
@@ -73,45 +73,41 @@ bool gmenu_is_active()
...
 {
-	int step, steps;
-
 	if ((sgpCurrItem->dwFlags & GMENU_SLIDER) == 0)
 		return;
 
-	step = sgpCurrItem->dwFlags & 0xFFF;
-	steps = (int)(sgpCurrItem->dwFlags & 0xFFF000) >> 12;
+	int step = sgpCurrItem->dwFlags & 0xFFF;
+	int steps = (int)(sgpCurrItem->dwFlags & 0xFFF000) >> 12;
```
Comment (thebigMuh) [inline Source/gmenu.cpp:110]: The (int) is of course of the devil and will cause nasal demons eventually. Don't know why it's even there.

There are multiple (int)s in this review.

File: Source/stores.cpp (hunk 6 above)
Comment (AJenbo) [reply Source/stores.cpp:193]: magic can exceed 128, afaik it's not safe to simply change the two others to unsigned.

File: Source/gmenu.cpp (hunk 8 above)
Comment (AJenbo) [reply Source/gmenu.cpp:110]: Since I didn't add them they are out of scope for this review.

File: Source/stores.cpp (hunk 7 above)
Comment (AJenbo) [reply Source/stores.cpp:452]: also std::max(1)

File: Source/inv.cpp (hunk 5 above)
Comment (AJenbo) [reply Source/inv.cpp:1945]: PlayerStruct &myPlayer = plr[myplr];
even

File: Source/stores.cpp (hunk 6 above)
Comment (thebigMuh) [reply Source/stores.cpp:193]: Scary

**Comment by julealgon** [comment]
Resolve conflicts please.

**Comment by AJenbo** [comment]
> Resolve conflicts please.

No I refuse

**Comment by AJenbo** [comment]
> Resolve conflicts please.

Seriously though don't you think you can review the changes, at least generally, considering that the only conflict is that a couple of functions where renamed by you and Glebm.

**Comment by julealgon** [comment]
> Seriously though don't you think you can review the changes, at least generally, considering that the only conflict is that a couple of functions where renamed by you and Glebm.

I have no idea what the conflicts are (nor do I think I should need to know). I usually only review things after conflicts are resolved so that I don't have to re-review everything again once the changes are rebased.

File: Source/inv.cpp [hunk 9]
```diff
@@ -1504,117 +1444,125 @@ void CheckInvScrn(bool isShiftHeld)
...
-			tmp = items[item_num];
-			memset(&items[item_num], 0, sizeof(*items));
-			GetItemAttrs(item_num, IDI_FULLNOTE, 16);
-			SetupItem(item_num);
-			plr[pnum].HoldItem = items[item_num];
-			items[item_num] = tmp;
-		}
-	}
+
+	CheckNaKrulNotes(player);
```
Comment (julealgon) [inline Source/inv.cpp:1565]: Nice split.

**Review by julealgon — APPROVED** [review]

**Pull request merged**